src/controller/RentalsController.java

```diff
@@ -168,4 +179,46 @@ public static Rental getRental(int rid) {
 
         return retRental;
     }
+
+
+    public static Vehicle rentVehicle(int confNo, String cardName, int cardNo, String expDate){
+        Reservation reservation = ReservationsController.getReservation(confNo);
+        String vtname, location, city;
+        ArrayList<Vehicle> availableVehicles;
+        String fromDate, fromTime, toDate, toTime;
+        Vehicle rentedVehicle = null;
+
+        vtname = reservation.getVtname();
+        location = reservation.getLocation();
+        city = reservation.getCity();
+        fromDate = reservation.getPickUpDate();
+        fromTime = reservation.getPickUpTime();
+        toDate = reservation.getDropOffDate();
+        toTime = reservation.getDropOffTime();
+
+        availableVehicles = VehiclesController.getAvailableVehicles(vtname, location, city);
+        if (availableVehicles != null){
+            rentedVehicle = availableVehicles.get(0);
+            VehiclesController.changeStatus(rentedVehicle.getVlicense(), "Rented");
+            Rental rental = new Rental(
```

Comment:
I think you forgot to call addRental on this rental

Reply:
btw, you should try adding the rental before changing the vehicle status, in case the rental fails (although we'll deal with errors later, but we should change the order so we don't forget)